src/McpServerService.cs

```diff
@@ -0,0 +1,192 @@
+        {
+            _cancellationTokenSource.Cancel();
+            _server?.Dispose();
+            _packageLoadedEvent.Dispose();
+        }
+
+        /// <summary>
+        /// Waits for the package data to be loaded and ready for serving.
+        /// This ensures we don't serve stale or incomplete data.
+        /// </summary>
+        /// <param name="cancellationToken">Token to cancel the wait operation</param>
+        /// <returns>A task that completes when package data is ready</returns>
+        private async Task WaitForPackageDataAsync(CancellationToken cancellationToken)
+        {
+            // Use TaskCompletionSource to wait for IsLoading to become false
+            var isLoadingTcs = new TaskCompletionSource<bool>();
+            
+            // Check if already not loading
+            if (MainWindow.Instance.IsLoading == false)
+            {
+                isLoadingTcs.SetResult(true);
+            }
+            else
+            {
+                // Listen for PropertyChanged event
+                PropertyChangedEventHandler handler = (sender, e) =>
+                {
+                    if (e.PropertyName == nameof(MainWindow.IsLoading) && 
+                        MainWindow.Instance.IsLoading == false)
+                    {
+                        isLoadingTcs.TrySetResult(true);
+                    }
+                };
+
+                MainWindow.Instance.PropertyChanged += handler;
+
+                // Ensure we clean up the event handler
+                cancellationToken.Register(() =>
+                {
+                    MainWindow.Instance.PropertyChanged -= handler;
+                    isLoadingTcs.TrySetCanceled();
+                });
+
+                // Wait for IsLoading to become false or cancellation
+                await isLoadingTcs.Task;
+
+                // Clean up the event handler
+                MainWindow.Instance.PropertyChanged -= handler;
+            }
+
+            // Additional safety check - ensure packages are actually loaded
+            while (PackageCatalogModel.Instance?.Packages == null || 
+                   PackageCatalogModel.Instance.Packages.Count == 0)
+            {
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    throw new OperationCanceledException();
+                }
+
+                await Task.Delay(100, cancellationToken);
+            }
+
+            _isPackageDataReady = true;
+            _packageLoadedEvent.Set();
+        }
+
+        /// <summary>
+        /// MCP Tool: Lists all Package Family Names from the loaded packages.
+        /// This tool provides access to the package family names of all installed packages
+        /// that have been loaded by the application.
+        /// </summary>
+        /// <returns>A list of package family names</returns>
+        [McpTool("list_package_family_names", "Lists all Package Family Names from loaded AppX packages")]
```

Comment:
AppX and MSIX are synonyms, so include both in the description to make it easier to find

```suggestion
        [McpTool("list_package_family_names", "Lists all Package Family Names from loaded MSIX (AppX) packages")]
``` #Resolved